code/modules/mob/living/carbon/human/species.dm

```diff
@@ -1703,6 +1703,21 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 				H.adjust_bodytemperature((thermal_protection+1)*natural + min(thermal_protection * (loc_temp - H.bodytemperature) / BODYTEMP_HEAT_DIVISOR, BODYTEMP_HEATING_MAX))
 			else //we're sweating, insulation hinders out ability to reduce heat - but will reduce the amount of heat we get from the environment
 				H.adjust_bodytemperature(natural*(1/(thermal_protection+1)) + min(thermal_protection * (loc_temp - H.bodytemperature) / BODYTEMP_HEAT_DIVISOR, BODYTEMP_HEATING_MAX))
+		switch((loc_temp - H.bodytemperature)*thermal_protection)
```

Comment:
I feel this could be better made than a super if chain. datums or sommat

Reply:
I couldnt really think of any other way I could implement it without a switch() statement